### PR TITLE
Add FormEncoded to support the slightly different encoding of x-www-f…

### LINKED ISSUE #4: Encoding of x-www-form-urlencoded broken
WwwFormUrlEncoded uses %20 to encode spaces when it should use "+".
Also it doesn't normalize new line sequences.

### PR DESCRIPTION
**Pull request by dmfs**
…orm-urlencoded parameters. Fixes #4

### CODE REVIEW

File: rfc3986-uri/src/main/java/org/dmfs/rfc3986/encoding/FormEncoded.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        try
+        {
+            // TODO: improve the performance. This appears to have quite some overhead compared to URLEncoder.encode().
+            OutputStream out = new FormPercentEncodingOutputStream(len, CharSets.UNRESERVED);
+            Writer w = new OutputStreamWriter(out, charSet);
+            w.append(charSequence);
+            w.close();
+            return out.toString();
+        }
+        catch (UnsupportedEncodingException e)
```
Comment (lemonboston) [inline rfc3986-uri/src/main/java/org/dmfs/rfc3986/encoding/FormEncoded.java:154]: This `catch` could be removed.

**Review by lemonboston — APPROVED** [review]
I've checked the changes but without the knowledge of the encoding specs, let me know I should get familiar with them now and get back to this for a thorough review.

File: rfc3986-uri/src/main/java/org/dmfs/rfc3986/encoding/FormEncoded.java (hunk 1 above)
Comment (dmfs) [reply rfc3986-uri/src/main/java/org/dmfs/rfc3986/encoding/FormEncoded.java:154]: `UnsupportedEncodingException` is a subclass of `IOException`, the separate catch-rethrow makes sure that unsupported encodings are thrown normally while other IOExceptions are re-thrown as RuntimeExceptions.

**Comment by dmfs** [comment]
No need to get familiar with the encoding specs for now.

File: rfc3986-uri/src/main/java/org/dmfs/rfc3986/encoding/FormEncoded.java (hunk 1 above)
Comment (lemonboston) [reply rfc3986-uri/src/main/java/org/dmfs/rfc3986/encoding/FormEncoded.java:154]: Oh, I see, I missed to notice that it is a subtype of `IOException`.

**Pull request merged**